flocker/restapi/_infrastructure.py

```diff
@@ -210,6 +222,22 @@ def deco(original):
         @_logging
         @_serialize(outputValidator)
         def loadAndDispatch(self, request, **routeArguments):
+            # If-Matches support:
+            if request.requestHeaders.hasHeader(b"if-matches"):
+                if get_etag is None:
+                    raise BadRequest(
+                        BAD_REQUEST,
+                        {u"description":
+                         u"If-Matches not supported by this method."})
+                etag = get_etag(self)
+                if_matches = request.requestHeaders.getRawHeaders(
+                    b"if-matches")
```

Comment:
I don't think this is parsing the headers correctly. The [RFC](https://tools.ietf.org/html/rfc7232#section-3.1) describes this header as a comma separated list of strings, with `"` surrounding the bodies.